File: TicketsSoporte/logica/FlujoTicket.cs
namespace TicketsSoporte.logica
{
    /// <summary>
    /// Define y valida el flujo de estados permitido de un ticket:
    /// Abierto -> Asignado -> (Escalado) -> Resuelto -> Cerrado.
    /// </summary>
    public class FlujoTicket
    {
        private readonly Dictionary<string, List<string>> dicTransiciones;

        public FlujoTicket()
        {
            dicTransiciones = new Dictionary<string, List<string>>
            {
                { "Abierto", new List<string> { "Asignado" } },
                { "Asignado", new List<string> { "Escalado", "Resuelto" } },
                { "Escalado", new List<string> { "Asignado", "Resuelto" } },
                { "Resuelto", new List<string> { "Cerrado", "Asignado" } },
                { "Cerrado", new List<string>() }
            };
        }

        public bool puedeCambiarEstado(string strEstadoActual, string strEstadoNuevo)
        {
            return dicTransiciones.ContainsKey(strEstadoActual) && dicTransiciones[strEstadoActual].Contains(strEstadoNuevo);
        }

        public void mostrarFlujo()
        {
            Console.WriteLine("\n=== FLUJO DE ESTADOS DEL TICKET ===");
            Console.WriteLine(" Abierto -> Asignado -> Escalado -> Resuelto -> Cerrado");
            Console.WriteLine("              \\<--------/  (reasignacion tras escalar)");
            Console.WriteLine("                           \\-> Asignado (reapertura)");
            foreach (KeyValuePair<string, List<string>> objPar in dicTransiciones)
            {
                string strDestinos = objPar.Value.Count > 0 ? string.Join(", ", objPar.Value) : "(estado final)";
                Console.WriteLine($" {objPar.Key} => {strDestinos}");
            }
        }
    }
}

File: BlueJ_Proyecto/FlujoTicket.java
import java.util.ArrayList;
import java.util.HashMap;

public class FlujoTicket
{
    private HashMap<String, ArrayList<String>> transiciones;

    public FlujoTicket()
    {
        transiciones = new HashMap<String, ArrayList<String>>();

        ArrayList<String> desdeAbierto = new ArrayList<String>();
        desdeAbierto.add("Asignado");
        transiciones.put("Abierto", desdeAbierto);

        ArrayList<String> desdeAsignado = new ArrayList<String>();
        desdeAsignado.add("Escalado");
        desdeAsignado.add("Resuelto");
        transiciones.put("Asignado", desdeAsignado);

        ArrayList<String> desdeEscalado = new ArrayList<String>();
        desdeEscalado.add("Asignado");
        desdeEscalado.add("Resuelto");
        transiciones.put("Escalado", desdeEscalado);

        ArrayList<String> desdeResuelto = new ArrayList<String>();
        desdeResuelto.add("Cerrado");
        desdeResuelto.add("Asignado");
        transiciones.put("Resuelto", desdeResuelto);

        transiciones.put("Cerrado", new ArrayList<String>());
    }

    public boolean puedeCambiarEstado(String estadoActual, String estadoNuevo)
    {
        return transiciones.containsKey(estadoActual) && transiciones.get(estadoActual).contains(estadoNuevo);
    }

    public void mostrarFlujo()
    {
        System.out.println("=== FLUJO DE ESTADOS DEL TICKET ===");
        System.out.println("Abierto -> Asignado -> Escalado -> Resuelto -> Cerrado");

        for (String estado : transiciones.keySet()) {
            ArrayList<String> destinos = transiciones.get(estado);
            String texto = destinos.isEmpty() ? "(estado final)" : String.join(", ", destinos);
            System.out.println(" " + estado + " => " + texto);
        }
    }
}
